### PR TITLE
Ar 97 en tant quutilisateur je shouaite pouvoir creer une area sur mobile

### PR DESCRIPTION
**Pull request by romainpanno**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [x] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**

* **Issue ticket number and link**

## Summary by CodeRabbit

- **New Features**
  - Added enhanced search functionality with user-specific results.
  - Introduced a new custom modal component for an improved user interface on mobile.
  - Implemented a service selection component to choose from different services more conveniently.
  - Added profile management and navigation features in the mobile app.

- **Improvements**
  - Improved the readability and usability of the mobile app by refactoring components to use styled components.
  - Updated modal behavior and visibility control for a smoother user experience.
  - Enhanced error handling and logging for better debugging and user feedback.

- **Bug Fixes**
  - Fixed issues with form data handling, error states, and submission logic in various mobile app components.
  - Removed unnecessary console logs and corrected logging levels for errors.

- **Documentation**
  - Updated `.gitignore` files to better manage environment configuration files and keep sensitive data secure.

- **Refactor**
  - Refactored several mobile app components to use arrow function syntax and promises for better code quality and performance.
  - Reorganized import statements and standardized the use of double quotes for consistency across files.

- **Style**
  - Applied consistent styling and code formatting across multiple mobile app components for better code readability and maintenance.

- **Chores**
  - Modified internal interfaces and class structures for more efficient data handling within the mobile app.

### CODE REVIEW

**Review by KitetsuK — CHANGES_REQUESTED** [review]

File: back/.gitignore [hunk 1]
```diff
@@ -156,3 +156,8 @@ docs/pages/versions/*/react-native/*.diff
 /ios/versioned-react-native/*/ReactNative/sdks/hermes-engine/destroot
 **/.env
 package-lock.json
+
+.env*
+.flaskenv*
+!.env.project
+!.env.vault
```
Comment (Vivelis) [inline back/.gitignore:163]: EOF

Comment (Vivelis) [inline back/.gitignore:163]: EOF

File: front/.gitignore [hunk 2]
```diff
@@ -35,5 +35,10 @@ yarn-error.log*
...
 next-env.d.ts
 package-lock.json
-.env.*
 .env
+
+.env.me
+.env.development
+.flaskenv*
+!.env.project
+!.env.vault
```
Comment (Vivelis) [inline front/.gitignore:44]: EOF

File: front/app/services/page.tsx [hunk 3]
```diff
@@ -105,9 +105,6 @@ export default function Services() {
    * @description useEffect to fetch actionJsonData when ModalData is defined
    */
   useEffect(() => {
```
Comment (Vivelis) [inline front/app/services/page.tsx:107]: useless useEffect(), should be removed

File: mobile/app/index.tsx [hunk 4]
```diff
@@ -11,16 +11,19 @@ const App = () => {
...
+      `${process.env.EXPO_PUBLIC_API_URL}/users/me`,
+      {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
       },
-    });
+    );
```
Comment (Vivelis) [inline mobile/app/index.tsx:23]: fetch should be used in a try-catch

**Review by Vivelis — CHANGES_REQUESTED** [review]

File: front/app/services/page.tsx (hunk 3 above)
Comment (romainpanno) [reply front/app/services/page.tsx:107]: We need it in order to tests sometimes

**Review by Vivelis — APPROVED** [review]
listen to the grand master codeRabbit

**Review by Vivelis — DISMISSED** [review]

**Review by KitetsuK — DISMISSED** [review]

File: mobile/app/profile.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      const response = await fetch(
+        process.env.EXPO_PUBLIC_API_URL + "/users/me",
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: "Bearer " + token,
+          },
+        },
+      );
```
Comment (Vivelis) [inline mobile/app/profile.tsx:42]: should catch errors of fetch

File: mobile/app/profile.tsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    const fetchData = async () => {
+      const response = await fetch(
+        process.env.EXPO_PUBLIC_API_URL + "/users/me",
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: "Bearer " + (await AsyncStorage.getItem("token")),
+          },
+        },
```
Comment (Vivelis) [inline mobile/app/profile.tsx:66]: should catch errors of fetch

**Review by Vivelis — CHANGES_REQUESTED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by badidu09 — APPROVED** [review]

**Pull request merged**